directives/submit.py

```diff
@@ -159,6 +163,25 @@ def run(self):
         if 'category' in self.options:
             data['category'] = str(self.options['category'])
 
+        source, line = self.state_machine.get_source_and_line(self.lineno)
+        if 'reveal-submission-feedback' in self.options:
+            data['reveal_submission_feedback'] = parse_reveal_rule(
+                self.options['reveal-submission-feedback'],
+                source,
+                line,
+                'reveal-submission-feedback',
+            )
+        if 'reveal-model-solutions' in self.options:
+            data['reveal_model_solutions'] = parse_reveal_rule(
+                self.options['reveal-model-solutions'],
+                source,
+                line,
+                'reveal-model-solutions',
+            )
+
+        if 'grading-mode' in self.options:
+            data['grading_mode'] = self.options['grading-mode']
```

Comment:
This should be done before the category override setting so that the `override` setting in `conf.py` can really override anything. Override is supposed to override even the settings defined directly in the submit directive options. (If override is used, people typically wouldn't define the overridden settings manually in the exercises, but any way, I would ensure that override really overrides anything.)

A few lines above, you can see the line `if category in override:`. You can move these lines just above it.

Reply:
Moved the lines